lib/view_model/active_record/update_operation.rb

```diff
@@ -421,114 +455,173 @@ def build_updates_for_collection_association(association_data, association_updat
         clear_association_cache(model, association_data.direct_reflection)
       end
 
-      child_datas =
-        case association_update
-        when OwnedCollectionUpdate::Replace
-          association_update.update_datas
+      # Update contents are either UpdateData in the case of a nested
```

Comment:
This type soup is very confusing. I _think_ it's correct since we have the guarantee of no duplicate references per fupdate, otherwise refs could only be moved once (`moved_refs` / `existing_refs` can't resolve strings to refs). It might help to define a `child_data_to_vm_ref` helper, then you can eliminate the conditionals mental overhead of dealing with strings, UpdateDatas, and nils.